app/assets/javascripts/code_editor.js

```diff
@@ -8,14 +8,50 @@ $(document).on('page:change', function(){
   editor.setShowPrintMargin(false);
   document.getElementById('wrapper').style.fontSize='15px';
 
-  // Updating local Storage
+   // Updating local Storage
   if(localStorage)
-    {
-      if(localStorage.getItem(localStorageID)){
-        editor.setValue(localStorage.getItem(localStorageID));
+  {
+    if(localStorage.getItem(localStorageID)){
+      editor.setValue(localStorage.getItem(localStorageID));
+    }
+    editor.getSession().on('change', function(e) {
+      localStorage.setItem(localStorageID, editor.getValue());
+    });
+  }
+
+  $(".submit").click( function() {
+
+    var expected_words = JSON.parse($("#key_words").text());
+    var answer = localStorage.getItem(localStorageID);
+    var is_valid_answer = true;
+
+    if (answer) {
+      for (var word in expected_words) {
+        var reg_exp = new RegExp(word , 'g')
+        var count = (answer.match(reg_exp) || []).length;
+        if (expected_words[word] != count) {
+          is_valid_answer = false;
+        }
       }
-      editor.getSession().on('change', function(e) {
-        localStorage.setItem(localStorageID, editor.getValue());
-      });
     }
+    else {
+      is_valid_answer = false;
+    }
+
+    $("#message").show();
+
+    if(!is_valid_answer) {
+      $("#fail").show();
+    }
+    else {
+       $("#success").show();
+       $.ajax({ url: window.location.href, method: "PATCH" });
+    }
+  });
+
+  $("#close").click( function() {
+    $("#fail").hide();
```

Comment:
You can include multiple selectors in one go I think, try `$("#fail, #success, #message").hide();`